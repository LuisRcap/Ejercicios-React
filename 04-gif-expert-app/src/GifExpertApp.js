import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);

    const onAddCategory = ( newCategory ) => {
        if ( categories.includes( newCategory ) ) return
        setCategories([ newCategory, ...categories ]);
    }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory 
                //setCategories={ setCategories }
                onNewCategory={ (value) => onAddCategory(value) }
            />
            <hr />

            <ol>
                {
                    categories.map( (category, i) => (
                        <GifGrid
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>
        </>
    );
};

export default GifExpertApp;