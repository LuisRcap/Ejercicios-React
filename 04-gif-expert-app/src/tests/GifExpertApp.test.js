import { render, screen } from '@testing-library/react';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {

    test('Debe de mostrar el componente correctamente', () => {
        const { container } = render( <GifExpertApp /> );
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorÃ­as', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const { container } = render(<GifExpertApp defaultCategories={ categories } />);

        expect( container.innerHTML ).toContain('One Punch');
        expect( container.innerHTML ).toContain('Dragon Ball');
        

    });
    
    
});