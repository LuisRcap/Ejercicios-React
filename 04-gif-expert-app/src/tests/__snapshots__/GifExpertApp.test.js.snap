// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar el componente correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <form
    aria-label="form"
  >
    <p />
    <input
      id="gif-input"
      type="text"
      value=""
    />
    <button
      class="btn"
      type="submit"
    >
      Agregar
    </button>
  </form>
  <hr />
  <ol />
</div>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categor√≠as 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <form
    aria-label="form"
  >
    <p />
    <input
      id="gif-input"
      type="text"
      value=""
    />
    <button
      class="btn"
      type="submit"
    >
      Agregar
    </button>
  </form>
  <hr />
  <ol>
    <h3
      class="animate__animated animate__fadeIn"
    >
      One Punch
    </h3>
    <p
      class="animate__animated animate__flash"
    >
      Loading
    </p>
    <div
      class="card-grid"
    />
    <h3
      class="animate__animated animate__fadeIn"
    >
      Dragon Ball
    </h3>
    <p
      class="animate__animated animate__flash"
    >
      Loading
    </p>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
